wandb/util.py

```diff
@@ -973,16 +975,35 @@ def _prompt_choice():
         return -1
 
 
-def prompt_choices(choices, allow_manual=False):
+def _prompt_choice_with_timeout(input_timeout=None):
+    timeout_log = (
+        "Input timeout occured! run `wandb login` to enable logging runs to W&B"
+    )
+    try:
+        choice = stdin_timeout(
+            prompt="%s: Enter your choice: " % term.LOG_STRING,
+            timeout=input_timeout,
+            timeout_log=timeout_log,
+        )
+        return int(choice) - 1
+    except ValueError:
+        return -1
+    except InputTimeoutError as timeoutError:
+        wandb.termlog(timeoutError.message)
+        return TIMEOUT_CODE
+
+
+def prompt_choices(choices, allow_manual=False, input_timeout=None):
```

Comment:
To minimize the risk of the funky signal logic breaking users environments, let's just use the old logic when input_timeout is None.

Reply:
done